pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -877,6 +894,8 @@ async def _call_tools(
     user_parts_by_index: dict[int, _messages.UserPromptPart] = {}
     deferred_calls_by_index: dict[int, Literal['external', 'unapproved']] = {}
 
+    _check_tool_call_limits(tool_manager, tool_calls, usage_limits)
```

Comment:
Now that we check this here, we don't need the check inside `ToolManager._call_tool` anymore, right?

Reply:
Also, can we pass `usage` into this `_call_tools` function (`ctx.state.usage`) and pass that on, instead of getting it from the `tool_manager`?

Reply:
Also, looks like `UsageLimits` will actually never be `None`, so we can remove `| None` and the checks we have